python/python/glide/async_commands/core.py

```diff
@@ -1324,6 +1428,124 @@ async def zcount(
+    ) -> List[str]:
+        """
+        Returns the specified range of elements in the sorted set stored at `key`.
+
+        ZRANGE can perform different types of range queries: by index (rank), by the score, or by lexicographical order.
+
+        See https://redis.io/commands/zrange/ for more details.
+
+        Args:
+            key (str): The key of the sorted set.
+            range_query (Union[RangeByIndex, RangeByLex, RangeByScore]): The range query object representing the type of range query to perform.
+                - For range queries by index (rank), use RangeByIndex.
+                - For range queries by lexicographical order, use RangeByLex.
+                - For range queries by score, use RangeByScore.
+            reverse (bool): If True, reverses the sorted set, with index 0 as the element with the highest score
+                By default, `start` must be less than or equal to `stop` to return anything.
+                However, if range_query is not an index query, the `start` is the highest score to consider, and `stop` is the lowest score to consider,
+                therefore `start` must be greater than or equal to `stop` in order to return anything.
+
+        Returns:
+            List[str]: A list of elements within the specified range.
+            For idex queries, If `start` is greater than either the end index of the sorted set or `stop`, an empty list is returned.
+            For lex/score queries:
+                - If `start` is greater than either the end index of the sorted set or `stop` and `reverse` is set to False,
+                 an empty list is returned.
+                - If `stop` is greater than either the end index of the sorted set or `start` and `reverse` is set to True,
+                 an empty list is returned.
+            If `key` does not exist, it is treated as an empty sorted set, and the command returns an empty array.
+            If `key` holds a value that is not a sorted set, an error is returned.
+
+        Examples:
+            >>> await zrange("my_sorted_set", RangeByIndex(0, -1))
+                ['member1', 'member2', 'member3']  # Returns all members in ascending order.
+            >>> await zrange("non_existing_key", RangeByScore(ScoreBoundary(0), ScoreBoundary(-1)))
+                []
+        """
+        args = [key, str(range_query.start), str(range_query.stop)]
+        if reverse:
+            args.append("REV")
+        if not isinstance(range_query, RangeByIndex):
+            args.append("BYSCORE" if type(range_query) is RangeByScore else "BYLEX")
+            if range_query.limit:
+                args.extend(
+                    [
+                        "LIMIT",
+                        str(range_query.limit.offset),
+                        str(range_query.limit.count),
+                    ]
+                )
+
+        return cast(List[str], await self._execute_command(RequestType.Zrange, args))
+
+    async def zrange_withscores(
```

Comment:
This function implementation is almost identical to zrange. I agree that it might give a better user experience to separate them in terms of the return value, and it would work better with java - but please move the mutual parts of the function to a separate function and call it from both. e.g. _create_zrange_args